lib/logstash/filters/aggregate.rb

```diff
@@ -195,6 +195,27 @@ class LogStash::Filters::Aggregate < LogStash::Filters::Base
   # Example value : `"map['sql_duration'] += event['duration']"`
   config :code, :validate => :string, :required => true
 
+
+
+  # This configuration describes the code block that is executed either on timeout 
+  # or in case 'push_previous_map_as_event' is set to true.
+  #
+  # Example value : `"map['sql_duration'] += event['duration']"`
+  config :timeout_code, :validate => :string, :required => false
+
+
+  # This sets the event's key for the task id value. The aggregate plugin maps the aggregation map to 
+  # a task_id value. This task_id will then be set into the event. This can help correlate which events have been timed out
+  #
+  # Example:
+  #
+  # If the taskId is "12345" and this field is set to "my_Id", the generated event will have:
+  # event[ "my_Id" ] = "12345"
+  #
+  # Default value: "task_id"
+  config :timeout_task_id_field, :validate => :string, :default => "task_id"
+
+
```

Comment:
Up to me, no default value for this option.
So that, by default, no task_id field is set on timeout event.
It would be nice to tell that in documentation.